client-go/controller/client/auth.go

```diff
@@ -0,0 +1,243 @@
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/deis/deis/client-go/controller/api"
+)
+
+// Register a user with the controller
+func Register(URL url.URL, username string, password string, email string, sslVerify bool,
+	loginAfter bool) error {
+	controllerURL := URL
+	client := CreateHTTPClient(sslVerify)
+
+	// Create a JSON body of request from user api
+	user := api.AuthRegisterRequest{Username: username, Password: password, Email: email}
+	body, err := json.Marshal(user)
+
+	if err != nil {
+		return err
+	}
+
+	URL.Path = "/v1/auth/register/"
+
+	headers := http.Header{}
+
+	controllerClient, err := Load()
+
+	if err == nil {
+		headers.Add("Authorization", "token "+controllerClient.Token)
+	}
+
+	headers.Add("Content-Type", "application/json")
+	addUserAgent(&headers)
+
+	res, err := rawRequest(client, "POST", URL.String(), bytes.NewBuffer(body), headers, 201)
+
+	if err != nil {
+		return err
+	}
+
+	res.Body.Close()
```

Comment:
I think it would be more defensive coding to move this above the error check and call it as `defer res.Body.Close()`.